File: Lib/Telerik/Examples/QuickStart/VirtualGrid/SaveLoadLayout/Customer.cs
using System;

namespace Telerik.Examples.WinControls.VirtualGrid.SaveLoadLayout
{
    public class Customer
    {
        public string CustomerId { get; set; }
        public string CompanyName { get; set; }
        public string ContactName { get; set; }
        public string Country { get; set; }
        public string City { get; set; }

        public Customer(string customerId, string companyName, string contactName, string country, string city)
        {

            this.CustomerId = customerId;
            this.CompanyName = companyName;
            this.ContactName = contactName;
            this.Country = country;
            this.City = city;
        }

        public string this[int i]
        {
            get
            {
                switch (i)
                {
                    case 0: return CustomerId;
                    case 1: return CompanyName;
                    case 2: return ContactName;
                    case 3: return Country;
                    case 4: return City;
                    default: return String.Empty;
                }
            }
        }
    }
}

File: Lib/Telerik/Examples/QuickStart/VirtualGrid/SaveLoadLayout/Customer.vb


Namespace Telerik.Examples.WinControls.VirtualGrid.SaveLoadLayout
	Public Class Customer
		Public Property CustomerId() As String
			Get
				Return m_CustomerId
			End Get
			Set
				m_CustomerId = Value
			End Set
		End Property
		Private m_CustomerId As String
		Public Property CompanyName() As String
			Get
				Return m_CompanyName
			End Get
			Set
				m_CompanyName = Value
			End Set
		End Property
		Private m_CompanyName As String
		Public Property ContactName() As String
			Get
				Return m_ContactName
			End Get
			Set
				m_ContactName = Value
			End Set
		End Property
		Private m_ContactName As String
		Public Property Country() As String
			Get
				Return m_Country
			End Get
			Set
				m_Country = Value
			End Set
		End Property
		Private m_Country As String
		Public Property City() As String
			Get
				Return m_City
			End Get
			Set
				m_City = Value
			End Set
		End Property
		Private m_City As String

		Public Sub New(customerId As String, companyName As String, contactName As String, country As String, city As String)

			Me.CustomerId = customerId
			Me.CompanyName = companyName
			Me.ContactName = contactName
			Me.Country = country
			Me.City = city
		End Sub

		Public Default ReadOnly Property Item(i As Integer) As String
			Get
                Select Case i
                    Case 0
                        Return CustomerId
                    Case 1
                        Return CompanyName
                    Case 2
                        Return ContactName
                    Case 3
                        Return Country
                    Case 4
                        Return City
                    Case Else
                        Return [String].Empty
                End Select
			End Get
		End Property
	End Class
End Namespace